internal/client.go

```diff
@@ -626,6 +630,515 @@ type (
+		// 			{
+		// 				Start: 1,
+		// 				End:   12,
+		// 			},
+		// 		},
+		// 		DayOfWeek: client.ScheduleRanges{
+		// 			{
+		// 				Start: 1,
+		//				End: 3,
+		// 			},
+		// 			{
+		// 				Start: 5,
+		// 			},
+		// 		},
+		// 	}
+		//
+		//
+		// The string can have 5, 6, or 7 fields, separated by spaces, and they are interpreted in the
+		// same way as a ScheduleCalendarSpec:
+		//	- 5 fields:         Minute, Hour, DayOfMonth, Month, DayOfWeek
+		//	- 6 fields:         Minute, Hour, DayOfMonth, Month, DayOfWeek, Year
+		//	- 7 fields: Second, Minute, Hour, DayOfMonth, Month, DayOfWeek, Year
+		//
+		// Notes:
+		//	- If Year is not given, it defaults to *.
+		//	- If Second is not given, it defaults to 0.
+		//	- Shorthands @yearly, @monthly, @weekly, @daily, and @hourly are also
+		//		accepted instead of the 5-7 time fields.
+		//	- @every <interval>[/<phase>] is accepted and gets compiled into an
+		//		IntervalSpec instead. <interval> and <phase> should be a decimal integer
+		//		with a unit suffix s, m, h, or d.
+		//	- Optionally, the string can be preceded by CRON_TZ=<timezone name> or
+		//		TZ=<timezone name>, which will get copied to ScheduleSpec.TimeZone. (In which case the ScheduleSpec.TimeZone field should be left empty.)
+		//	- Optionally, "#" followed by a comment can appear at the end of the string.
+		//	- Note that the special case that some cron implementations have for
+		//		treating DayOfMonth and DayOfWeek as "or" instead of "and" when both
+		//		are set is not implemented.
+		CronExpressions []string
+
+		// Skip - Any matching times will be skipped.
+		//
+		// All aspects of the schedulepb.ScheduleCalendarSpec—including seconds—must match a time for the time to be skipped.
+		Skip []ScheduleCalendarSpec
+
+		// StartAt - Any times before `startAt` will be skipped. Together, `startAt` and `endAt` make an inclusive interval.
+		// Optional: Defaulted to the beginning of time
+		StartAt *time.Time
+
+		// EndAt - Any times after `endAt` will be skipped.
+		// Optional: Defaulted to the end of time
+		EndAt *time.Time
+
+		// Jitter - All times will be incremented by a random value from 0 to this amount of jitter.
+		// Optional: Defaulted to 0
+		Jitter time.Duration
+
+		// Timezone - IANA timezone name, for example `US/Pacific`.
+		// Optional: Defaulted to UTC
+		Timezone time.Location
+	}
+
+	// ScheduleAction represents an action a schedule can take.
+	ScheduleAction interface {
+		isScheduleAction()
+	}
+
+	// ScheduleWorkflowAction implements ScheduleAction to launch a workflow.
+	ScheduleWorkflowAction struct {
+		// Workflow - What workflow to run.
+		Workflow interface{}
+
+		// Args - Arguments to pass to the workflow.
+		Args []interface{}
+
+		// WorkflowOptions - Options to use when launching the workflow.
+		WorkflowOptions StartWorkflowOptions
+	}
+
+	// ScheduleWorkflowActionDescription implements ScheduleAction to launch a workflow.
+	ScheduleWorkflowActionDescription struct {
```

Comment:
We need a separate types for the creation and description of a workflow action because of the `Workflow `, `Args`, and the `Memo ` and `SearchAttributes ` of the workflow

Reply:
>It's confusing that I can create a schedule with a description action.

it's a description of an action, I think it is pretty normal that a description of an object could be used to create a new one. A potential use case would be if you want to copy a schedule action with a different schedule spec or to a new namespace.

Another option I thought of is remove the type `ScheduleWorkflowAction` and replace with a function 
```
CreateScheduleWorkflowAction(Workflow interface{}, Args []interface{}, WorkflowOptions StartWorkflowOptions) ScheduleWorkflowActionDescription
```

`ScheduleWorkflowActionDescription` could then be renamed

Reply:
Having two types of actions represent the same thing on the same interface is rough for general-purpose/helper code. We should think about possible alternatives. Definitely want simple to create and simple to update.